From fbc590c77f5028ff690ed707bdf93c4d3b8445f3 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Tue, 15 Apr 2008 11:11:38 +0800
Subject: [PATCH] ENGR00072220 IMX31 Display error at the top of LCD

Top 10 lines of LCD display as black.
The upper-margin of the LCD must set 0 for iMX31,
because the Data Enable signal is always true on
iMX31 3ds.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/video/mxc/mxcfb_epson_vga.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/drivers/video/mxc/mxcfb_epson_vga.c b/drivers/video/mxc/mxcfb_epson_vga.c
old mode 100644
new mode 100755
index 0ef6bb8..07899c4
--- a/drivers/video/mxc/mxcfb_epson_vga.c
+++ b/drivers/video/mxc/mxcfb_epson_vga.c
@@ -36,6 +36,7 @@
 #include <linux/regulator/regulator.h>
 #include <linux/spi/spi.h>
 #include <asm/arch/mxcfb.h>
+#include <asm/mach-types.h>
 
 static struct spi_device *lcd_spi;
 
@@ -50,7 +51,7 @@ static struct regulator *core_reg;
 static struct fb_videomode video_modes[] = {
 	{
 	 /* 480x640 @ 60 Hz */
-	 "Epson-VGA", 60, 480, 640, 41701, 0, 41, 10, 5, 20, 10,
+	 "Epson-VGA", 60, 480, 640, 41701, 60, 41, 10, 5, 20, 10,
 	 FB_SYNC_CLK_INVERT | FB_SYNC_OE_ACT_HIGH,
 	 FB_VMODE_NONINTERLACED,
 	 0,},
@@ -64,6 +65,11 @@ static void lcd_init_fb(struct fb_info *info)
 
 	fb_videomode_to_var(&var, &video_modes[0]);
 
+	if (machine_is_mx31_3ds()) {
+		var.upper_margin = 0;
+		var.left_margin = 0;
+	}
+
 	var.activate = FB_ACTIVATE_ALL;
 	var.yres_virtual = var.yres * 2;
 
-- 
1.5.4.4

